metadata/metadata.go

```diff
@@ -156,28 +188,34 @@ var bannedStrings = [...]string{"__"}
 var bannedPrefixes = [...]string{"_"}
 var bannedSuffixes = [...]string{"_"}
 
-func resourceNamedSafely(id ResourceID) error {
+func resourceNamedSafely(id ResourceID, logger logging.Logger) error {
 	for _, substr := range bannedStrings {
 		if strings.Contains(id.Name, substr) {
+			logger.Errorw("Name contains banned string", "name", id.Name, "banned substring", substr)
```

Comment:
So what you should be doing for all these in the future is

err := fferr.NewInvalidResource...
logger.Errorw("Name contains banned string", "err", err)